Proxysql service crash/restart when monitor get aws aurora auto-discovered master endpoint failures

- [x] A clear description of the issue
   We have proxysql latest 3.0.5 version setup in kubernetes with aws aurora mysql and auto-discovery enabled, when aurora get slow and monitor checks fail this bring proxysql complete failure and exit/shutdown with `Error: signal 11` (see the attached logs). The aurora writes endpoint my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com is set as seed server in shunned state.
- [x] ProxySQL version
   3.0.5
- [x] OS version
   proxysql provisioned in kubernetes(aws eks AL2023 nodes) pod based on dockerhub official image proxysql/proxysql
- [x] The steps to reproduce the issue
   the issue reproduces when there are proxysql monitor timeouts on aurora master health-checks(see the attached logs)
- [x] The package used to install ProxySQL
   we use proxysql official dockerhub image: proxysql/proxysql:3.0.5
- [x] The **full** ProxySQL error log (default location: `/var/lib/proxysql/proxysql.log`)
   
```log
2026-02-08 19:05:04 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.6.230:51742
2026-02-08 19:05:04 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.12.22:38594
2026-02-08 19:05:04 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.7.231:49048
2026-02-08 19:05:04 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.12.22:38588
2026-02-08 19:05:04 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.6.230:51748
2026-02-08 19:05:05 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.11.5:51730
2026-02-08 19:05:05 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.8.219:51020
2026-02-08 19:05:05 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.6.53:38668
2026-02-08 19:05:05 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.9.205:36942
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.26.145:50496
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.7.53:48182
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.2.105:57236
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.13.240:37736
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.9.66:55660
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.1.232:58878
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.11.68:54848
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.14.58:45728
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.7.231:49062
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.11.5:51734
2026-02-08 19:05:06 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.26.145:50500
2026-02-08 19:05:07 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.26.145:50498
2026-02-08 19:05:07 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.14.170:49538
2026-02-08 19:05:07 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.5.200:38670
2026-02-08 19:05:07 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.8.219:51022
2026-02-08 19:05:08 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.25.184:41696
2026-02-08 19:05:08 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.4.56:36538
2026-02-08 19:05:08 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.26.145:50502
2026-02-08 19:05:08 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.1.22:40726
2026-02-08 19:05:09 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.8.212:35954
2026-02-08 19:05:12 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.1.50:40716
2026-02-08 19:05:12 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 504ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:12 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 504ms on server my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:13 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 505ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:13 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 505ms on server my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:13 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 504ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:13 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 504ms on server my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:15 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 506ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:15 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 506ms on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:15 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.1.22:40734
2026-02-08 19:05:16 mysql_connection.cpp:1343:handler(): [ERROR] Connect timeout on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : exceeded by 6us
2026-02-08 19:05:16 mysql_connection.cpp:1343:handler(): [ERROR] Connect timeout on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : exceeded by 9us
2026-02-08 19:05:16 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:16 MySQL_Session.cpp:1612:handler_again___status_PINGING_SERVER(): [ERROR] Ping timeout during ping on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 700003us (timeout 700ms)
2026-02-08 19:05:16 MySQL_Session.cpp:1612:handler_again___status_PINGING_SERVER(): [ERROR] Ping timeout during ping on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 700003us (timeout 700ms)
2026-02-08 19:05:17 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.26.145:43728
2026-02-08 19:05:17 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.8.219:57130
2026-02-08 19:05:17 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 507ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:17 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 507ms on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:18 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.9.66:37488
2026-02-08 19:05:18 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:19 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:21 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 508ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:21 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 508ms on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:21 MySQL_Session.cpp:1612:handler_again___status_PINGING_SERVER(): [ERROR] Ping timeout during ping on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 700007us (timeout 700ms)
2026-02-08 19:05:22 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:22 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.15.39:45966
2026-02-08 19:05:23 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.1.22:41046
2026-02-08 19:05:23 MySQL_Monitor.cpp:1527:monitor_ping_thread(): [ERROR] Error on server my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout
2026-02-08 19:05:23 MySQL_Monitor.cpp:1527:monitor_ping_thread(): [ERROR] Error on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout
2026-02-08 19:05:23 mysql_connection.cpp:1343:handler(): [ERROR] Connect timeout on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : exceeded by 6us
2026-02-08 19:05:23 mysql_connection.cpp:1343:handler(): [ERROR] Connect timeout on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : exceeded by 4us
2026-02-08 19:05:23 mysql_connection.cpp:1343:handler(): [ERROR] Connect timeout on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : exceeded by 2us
2026-02-08 19:05:23 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.7.231:60806
2026-02-08 19:05:23 mysql_connection.cpp:1343:handler(): [ERROR] Connect timeout on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : exceeded by 7us
2026-02-08 19:05:25 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:26 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1002ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:28 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 505ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:28 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 505ms on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:29 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:29 MySQL_Monitor.cpp:4774:monitor_dns_resolver_thread(): [ERROR] An error occurred while resolving hostname: application-autoscaling-def5d46e-e7ee-4074-a6a2-fe8f9cb6ed18.cnu0i28c0lxt.eu-central-1.rds.amazonaws.com [-2]
2026-02-08 19:05:30 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.2.93:56824
2026-02-08 19:05:30 MySQL_Monitor.cpp:4774:monitor_dns_resolver_thread(): [ERROR] An error occurred while resolving hostname: application-autoscaling-f5dc4ced-492d-4b0c-a2b8-645e09a0424e.cnu0i28c0lxt.eu-central-1.rds.amazonaws.com [-2]
2026-02-08 19:05:30 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.26.145:42366
2026-02-08 19:05:30 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.15.247:43434
2026-02-08 19:05:31 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.11.197:47822
2026-02-08 19:05:31 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 505ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:31 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 505ms on server my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:32 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 502ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:32 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 502ms on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:32 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 504ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:32 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 504ms on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:33 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.24.113:49786
2026-02-08 19:05:33 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.4.70:34270
2026-02-08 19:05:33 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.6.32:55538
2026-02-08 19:05:33 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.15.247:43456
2026-02-08 19:05:33 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.6.5:51866
2026-02-08 19:05:33 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.15.34:49876
2026-02-08 19:05:33 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.1.100:59298
2026-02-08 19:05:33 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.4.56:43398
2026-02-08 19:05:34 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.2.30:53920
2026-02-08 19:05:34 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 505ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:34 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 505ms on server my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:35 MySQL_Session.cpp:1612:handler_again___status_PINGING_SERVER(): [ERROR] Ping timeout during ping on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 700007us (timeout 700ms)
2026-02-08 19:05:35 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:35 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.5.200:53298
2026-02-08 19:05:35 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.9.161:50588
2026-02-08 19:05:36 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.1.50:51472
2026-02-08 19:05:37 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.5.200:40180
2026-02-08 19:05:37 MySQL_Session.cpp:1612:handler_again___status_PINGING_SERVER(): [ERROR] Ping timeout during ping on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 700005us (timeout 700ms)
2026-02-08 19:05:37 MySQL_Session.cpp:1612:handler_again___status_PINGING_SERVER(): [ERROR] Ping timeout during ping on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 700005us (timeout 700ms)
2026-02-08 19:05:37 MySQL_Session.cpp:1612:handler_again___status_PINGING_SERVER(): [ERROR] Ping timeout during ping on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 700005us (timeout 700ms)
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.10.55:42418
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.15.34:47326
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.24.113:45638
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.9.205:60492
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.14.58:39496
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.2.30:53958
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.9.66:41586
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.10.242:47648
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.12.42:59404
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.7.117:53932
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.19.125:37772
2026-02-08 19:05:37 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.2.30:41400
2026-02-08 19:05:38 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 504ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:38 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 504ms on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:39 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.6.32:44830
2026-02-08 19:05:39 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.4.56:44684
2026-02-08 19:05:39 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:40 mysql_connection.cpp:1343:handler(): [ERROR] Connect timeout on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 : exceeded by 9us
2026-02-08 19:05:40 MySQL_Thread.cpp:4028:ProcessAllSessions_Healthy0(): [WARNING] Closing unhealthy client connection 10.18.11.197:41436
2026-02-08 19:05:41 MySQL_Monitor.cpp:6321:monitor_AWS_Aurora_thread_HG(): [ERROR] Timeout on AWS Aurora health check for my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 504ms. If the server is overload, increase mysql_aws_aurora_hostgroups.check_timeout_ms
2026-02-08 19:05:41 MySQL_Monitor.cpp:6460:monitor_AWS_Aurora_thread_HG(): [ERROR] Error after 504ms on server my-rds.cluster-xxxxxx.eu-central-1.rds.amazonaws.com:3306 : timeout check
2026-02-08 19:05:42 MySQL_Monitor.cpp:6267:monitor_AWS_Aurora_thread_HG(): [ERROR] Error on AWS Aurora check for my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 1001ms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: timeout or error in creating new connection: Lost connection to server at 'handshake: reading initial communication packet', system error: 110.
2026-02-08 19:05:42 MySQL_Session.cpp:1612:handler_again___status_PINGING_SERVER(): [ERROR] Ping timeout during ping on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com:3306 after 700014us (timeout 700ms)
2026-02-08 19:05:42 MySQL_Session.cpp:1683:handler_again___status_RESETTING_CONNECTION(): [ERROR] Change user timeout during COM_CHANGE_USER on my-rds-master.xxxxxx.eu-central-1.rds.amazonaws.com , 3306
Error: signal 11:
proxysql(_Z13crash_handleri+0x2a) [0x56488954976a]
/lib/x86_64-linux-gnu/libc.so.6(+0x3fdf0) [0x7f7a80e2edf0]
proxysql(_ZN13MySQL_Session10RequestEndEP17MySQL_Data_StreamjPKc+0xc3) [0x5648895cf2b3]
proxysql(_ZN13MySQL_Session43handler_again___status_RESETTING_CONNECTIONEv+0x21f) [0x5648895dcf8f]
proxysql(_ZN13MySQL_Session7handlerEv+0x438) [0x5648895e4be8]
proxysql(_ZN12MySQL_Thread20process_all_sessionsEv+0x2be) [0x5648895c56ae]
proxysql(_ZN12MySQL_Thread3runEv+0x340) [0x5648895c5c70]
proxysql(_Z24mysql_worker_thread_funcPv+0x7c) [0x56488953365c]
/lib/x86_64-linux-gnu/libc.so.6(+0x92b7b) [0x7f7a80e81b7b]
/lib/x86_64-linux-gnu/libc.so.6(+0x1107b8) [0x7f7a80eff7b8]
 ---- proxysql(_Z13crash_handleri+0x2a) [0x56488954976a] : crash_handler(int)
 ---- proxysql(_ZN13MySQL_Session10RequestEndEP17MySQL_Data_StreamjPKc+0xc3) [0x5648895cf2b3] : MySQL_Session::RequestEnd(MySQL_Data_Stream*, unsigned int, char const*)
 ---- proxysql(_ZN13MySQL_Session43handler_again___status_RESETTING_CONNECTIONEv+0x21f) [0x5648895dcf8f] : MySQL_Session::handler_again___status_RESETTING_CONNECTION()
 ---- proxysql(_ZN13MySQL_Session7handlerEv+0x438) [0x5648895e4be8] : MySQL_Session::handler()
 ---- proxysql(_ZN12MySQL_Thread20process_all_sessionsEv+0x2be) [0x5648895c56ae] : MySQL_Thread::process_all_sessions()
 ---- proxysql(_ZN12MySQL_Thread3runEv+0x340) [0x5648895c5c70] : MySQL_Thread::run()
 ---- proxysql(_Z24mysql_worker_thread_funcPv+0x7c) [0x56488953365c] : mysql_worker_thread_func(void*)
To report a crashing bug visit: https://github.com/sysown/proxysql/issues
For support visit: https://proxysql.com/services/support/
```

--- a/lib/MySQL_Session.cpp
+++ b/lib/MySQL_Session.cpp
@@ -8883,7 +8883,9 @@ void MySQL_Session::RequestEnd(MySQL_Data_Stream *myds,const unsigned int myerrn
 	// @note This is a "temporary" solution that should be removed if packet queueing is implemented, since
 	// it will make the 'unexp_com_pings' field obsolete.
 	///////////////////////////////////////////////////////////////////////////////////////////////
-	if (client_myds->unexp_com_pings) {
+	// Fix #5355: Add null pointer check for client_myds to prevent use-after-free crash
+	// when session is deleted while RequestEnd() is still executing (e.g., during COM_CHANGE_USER timeout)
+	if (client_myds && client_myds->unexp_com_pings) {
 		client_myds->setDSS_STATE_QUERY_SENT_NET();
 
 		if (client_myds->unexp_com_pings) {

